### DISCUSSION TITLE
Build private repo on Docker

### DISCUSSION THREAD

**Discussion by nottatdat** [PyO3/maturin#1011] [Q&A]
I tried:

```
docker run --rm -v $(pwd):/io ghcr.io/pyo3/maturin build --release -i /opt/python/cp38-cp38/bin/python3.8
```

It got the following error:

```
Caused by:
  failed to authenticate when downloading repository

  * attempted ssh-agent authentication, but no usernames succeeded: `git`

  if the git CLI succeeds then `net.git-fetch-with-cli` may help here
  https://doc.rust-lang.org/cargo/reference/config.html#netgit-fetch-with-cli

Caused by:
  error authenticating: no auth sock variable; class=Ssh (23)
```

How to properly setup for these kinds of build?

**Comment by messense** [comment]
I've never used it with private repo before, but maybe you can refer to https://gist.github.com/d11wtq/8699521

**Comment by nottatdat** [comment]
Any idea on how to combine the two commands? I tried various ways but got no luck :(

**Reply by messense** [reply]
`docker run --rm -v $(dirname $SSH_AUTH_SOCK) -e SSH_AUTH_SOCK=$SSH_AUTH_SOCK -v $(pwd):/io ghcr.io/pyo3/maturin build --release -i /opt/python/cp38-cp38/bin/python3.8`

No idea if it's gonna work for you.
